internal/explorer/helpers.go

```diff
@@ -70,7 +72,7 @@ func (a *App) baseQuery(queryString string) (io.ReadCloser, error) {
 	if response.StatusCode != 200 {
 		var errResult interface{}
 		if err := json.NewDecoder(response.Body).Decode(errResult); err != nil {
-			return nil, fmt.Errorf("request failed: %w", errResult)
+			return nil, fmt.Errorf("request failed: %s", errResult)
```

Comment:
If the Decode method return an error, your errResult object will still be nil. what u probably meant to do is use `err` not `errResult` because that's what was returned by the Decode method.

In that case of course you should use the `%w` because it will wrap the error in the new error (which is a decoding error by the way, not a request failed)

```suggestion
			return nil, fmt.Errorf("failed to decode error from page: %w", err)
```